src/golang/cmd/server/handler/get_workflow_dag_result.go

```diff
@@ -153,10 +154,43 @@ func (h *GetWorkflowDagResultHandler) Perform(ctx context.Context, interfaceArgs
 		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving artifact results.")
 	}
 
+	contents, err := getArtifactContents(ctx, constructedDag, artifactResults)
+	if err != nil {
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving artifact contents.")
+	}
+
 	return dag.NewResultResponseFromDbObjects(
 		constructedDag,
 		dbWorkflowDagResult,
 		operatorResults,
 		artifactResults,
+		contents,
 	), http.StatusOK, nil
 }
+
+// TODO: We should replace this once we migrate to services based on `artifact` objects
+func getArtifactContents(
+	ctx context.Context,
+	dbWorkflowDag *workflow_dag.DBWorkflowDag,
+	dbArtifactResults []artifact_result.ArtifactResult,
+) (map[string]string, error) {
+	contents := make(map[string]string, len(dbArtifactResults))
+	storageObj := storage.NewStorage(&dbWorkflowDag.StorageConfig)
+	for _, artfResult := range dbArtifactResults {
+		if artf, ok := dbWorkflowDag.Artifacts[artfResult.ArtifactId]; ok {
+			// These artifacts has small content size and we can safely include them all in response.
```

Comment:
Should we directly check if the content size is below some threshold rather than the artifact type?

Reply:
Might be useful for string types since those are somewhat unbounded. I don't know if the additional complexity is worth it though. I'm sort of on the side of keeping the semantics simple for now, to avoid more confusion in the future.

Reply:
Yeah this is a good idea!